src/main/java/com/pingcap/tikv/operation/KVErrorHandler.java

```diff
@@ -84,7 +84,7 @@ public void handle(RespT resp) {
         regionManager.onRegionStale(ctxRegion.getId(), error.getStaleEpoch().getNewRegionsList());
         this.regionManager.onRegionStale(
             ctxRegion.getId(), error.getStaleEpoch().getNewRegionsList());
-        throw new StatusRuntimeException(Status.fromCode(Status.Code.CANCELLED).withDescription(error.toString()));
+        throw new StatusRuntimeException(Status.fromCode(Status.Code.UNAVAILABLE).withDescription(error.toString()));
```

Comment:
why modified both. you removed cancelled status from unrecoverable and change it to unavailable.
I think whether an exception is recoverable or not completely depend on TiKV's definition. It's better to confirm with the team.